src/sage/modules/free_module.py

```diff
@@ -7601,7 +7667,13 @@ def echelonized_basis(self):
             sage: V.span(B) == M
             True
         """
-        return self.__echelonized_basis
+        C = self.element_class
+        echelonized_basis = self.echelonized_basis_matrix().rows()
+        if echelonized_basis and echelonized_basis[0].parent() is not self:
+            w = [self(x) for x in echelonized_basis]
```

Comment:
```suggestion
        C = self.element_class
        echelonized_basis = self.echelonized_basis_matrix().rows()
        if echelonized_basis and echelonized_basis[0].parent() is not self:
            w = [C(self, x) for x in echelonized_basis]
```
If this doesn't work, `C` is unused and should be removed.

Reply:
in principle this ought to work, but apparently the lowermost function call (subclass constructor) expects a list instead of a vector, and expect the upper level to normalize it.

In retrospect we see the downside of "normalize as early as possible" paradigm:

* even if the caller want to bypass the logic upstream for performance (when they already know the desired parent etc.), they cannot do that without replicating part of the normalization logic;
* even when the downstream class want to implement special cases to speed up (e.g. copying from one `vector_complex_double_dense` to another ought to be much faster than copy it from and to a list), they cannot.

But that would be the topic of another pull request.